vdr/store/memory.go

```diff
@@ -109,26 +109,41 @@ func (m *memory) Resolve(id did.DID, metadata *vdr.ResolveMetadata) (*did.Docume
 	}
 
 	// return a deep copy
-	doc := &did.Document{}
-	docMetadata := &vdr.DocumentMetadata{}
+	copyEntry, err := deepCopy(entry)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &copyEntry.document, &copyEntry.metadata, nil
+}
 
+// deepCopy returns a deep copy of a memoryEntry
+func deepCopy(entry *memoryEntry) (*memoryEntry, error) {
```

Comment:
I thought it was going to be a function in `go-did`?

Reply:
go-did can do a proper memory based deepCopy. For now it's fixed here. An issue still has to be made on go-did.